crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -3551,6 +3667,26 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_read_only_tx_shared_lock() -> ResultTest<()> {
+        let (datastore, mut tx, table_id) = setup_table()?;
+        let row1 = u32_str_u32(1, "Foo", 18);
+        datastore.insert_mut_tx(&mut tx, table_id, row1.clone())?;
+        let row2 = u32_str_u32(2, "Bar", 20);
+        datastore.insert_mut_tx(&mut tx, table_id, row2.clone())?;
+        datastore.commit_mut_tx_for_test(tx)?;
+
+        // create multiple read only tx, and use them together.
+        let read_tx_1 = begin_tx(&datastore);
+        let read_tx_2 = begin_tx(&datastore);
+        let rows = vec![row1, row2];
+        assert_eq!(all_rows_tx(&read_tx_2, table_id), rows.clone());
+        assert_eq!(all_rows_tx(&read_tx_1, table_id), rows);
```

Comment:
```suggestion
        let rows = &[row1, row2];
        assert_eq!(&all_rows_tx(&read_tx_2, table_id), rows);
        assert_eq!(&all_rows_tx(&read_tx_1, table_id), rows);
```

Haven't tested this change, but my gut tells me it should work.